specifyweb/businessrules/views.py

```diff
@@ -1 +1,288 @@
+                            "description": "An object with keys corresponding to table names and values are an array of uniqueness rules",
+                            "additionalProperties": {
+                                "type": "array",
+                                "description": "The array of uniqueness rules for a given table",
+                                "items": UniquenessRuleSchema
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "put": {
+        "requestBody": {
+            "required": True,
+            "content": {
+                "application/json": {
+                    "schema": {
+                        "type": "object",
+                        "properties": {
+                            "rules": {
+                                "type": "array",
+                                "description": "The array of uniqueness rules for a given table",
+                                "items": UniquenessRuleSchema
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "responses": {
+            "201": {
+                "description": "Uniqueness rules properly updated and/or created",
+                "content": {
+                    "application/json": {
+                        "schema": {
+                            "type": "object"
+                        }
+                    }
+                }
+            }
+        }
+    }
+})
+@login_maybe_required
+@require_http_methods(['GET', 'PUT', 'POST'])
+@transaction.atomic
+def uniqueness_rule(request, discipline_id):
+    data = {}
+
+    try:
+        model = request.GET["model"]
+    except:
+        model = None
+
+    if request.method == 'GET':
+        rules = UniquenessRule.objects.filter(discipline=discipline_id)
+        for rule in rules:
+            rule_fields = rule.splocalecontaineritems.filter(
+                uniquenessrule_splocalecontaineritem__isScope=0)
+            scope = rule.splocalecontaineritems.filter(
+                uniquenessrule_splocalecontaineritem__isScope=1)
+
+            table = rule_fields[0].container.name
+            if model is not None and table.lower() != model.lower():
+                continue
+            if table not in data.keys():
+                data[table] = []
+            data[table].append({"rule": {"id": rule.id, "fields": [{"id": field.id, "name": field.name} for field in rule_fields], "scopes": [{
+                               "id": _scope.id, "name": _scope.name} for _scope in scope], "isDatabaseConstraint": rule.isDatabaseConstraint}})
+
+    elif request.method == 'PUT' or request.method == 'POST':
+        rules = json.loads(request.body)['rules']
+        discipline = models.Discipline.objects.get(id=discipline_id)
+        for rule in rules:
+            fetched_scopes = models.Splocalecontaineritem.objects.filter(
```

Comment:
Also assert that ids of fields all exist in the database. Throw a save conflict if not?

This, and the above, is done to make sure all fields exist in the database. Since you are using `id__in`, you might get a subset of them (because others could have been deleted by the time we handle it).